### ISSUE TITLE
Tesla M3 + GoodWe A-ES not completing handshake

### ISSUE THREAD

**Issue by gegere** [dalathegreat/Battery-Emulator#620]
### Expected Behavior

Currently on the [supported Inverter page](https://github.com/dalathegreat/Battery-Emulator/wiki#supported-inverters-list) the following support for Goodwe Inverters are listed. I am interested in helping to confirm is it will work with the following model: 

[**GoodWe A-ES Series**](https://en.goodwe.com/a-es-series-split-phase-hybrid-solar-inverter-for-north-america)
(Americas Only) 5-9.6 kW | Split Phase | Up to 4 MPPTs | Hybrid inverter (HV)

Would this work with the same parameters as **GoodWe ET/BT & EH/BH & EHB**?

### Settings

- Software version: ``
- Battery used: `Tesla M3 Standard Range`
- Inverter communication protocol: ``
- Hardware used for Battery-Emulator: `HW_LILYGO`
- CONTACTOR_CONTROL: `yes`
- DUAL_CAN: `yes`
- WEBSERVER: `yes`
- MQTT: ``

**Comment by dalathegreat** [comment]
Should work just fine, BYD is on the [approved list](https://www.invertersupply.com/media/data/GW_Approved%20Battery%20Options%20Statement-EN.pdf) for this inverter! Let us know how it goes!

**Comment by gegere** [comment]
Promising! Currently I am using ingenext Tesla Battery Controller and I am trying to resolve an issue charging on AC via the Tesla Mobile Charger OR j1772 Charger. The LilyGo RS485 Board will be arriving next week.

I am having an issue charging. The charge port will allow a charge a minute sometime up to 3 minuts, at 32 amps / 240VAC. 16 amps on each phase. Hopefully the **LilyGo RS485 / software** and this growing DIY battery community can help improve AC Charging and even DC/DC Fast Charging via the Tesla Charge Port.

Any tips OR thoughts on the situation?

![Screenshot 2024-11-17 at 9 15 07 PM](https://github.com/user-attachments/assets/070ac661-6772-416c-ab08-64eb266bfaed)

Then PCS_hvChargeStatue: PCS_CHARGE_BLOCKED


![Screenshot 2024-11-17 at 9 18 50 PM](https://github.com/user-attachments/assets/715fd7ec-e84f-4fd7-98ae-02bf8df3891d)

**Comment by dalathegreat** [comment]
Any progress on this testing @gegere ? :)

**Comment by gegere** [comment]
Yahoo! I have not yet purchased the **GoodWe A-ES Series Inverter**. I am currently in the process of adding a radiator to cool the Tesla M3 PCS/Battery during AC/DC charging using the charge port, with plans to complete this by January 10th. The system's features will include management using ESP32 Arduino microprocessors over CANBus.

The next project will involve removing the **Ingenext Tesla Battery Controller**, as their support has been terrible, plus their system is closed source. 

I'll be setting up the Battery Emulator on a Lily Board instead. I have successfully installed the Battery Emulator on the Lily Board without any issues and have the web server up and running! Exciting!

I truly appreciate your thoughts, time, and energy in helping move this great project forward. I will continue to be an active contributor going forward!

**Comment by gegere** [comment]
GW9600A-ES Inverter has arrived. Lily Board is setup, next step is get this all installed.

![Image](https://github.com/user-attachments/assets/c532c4d3-5ca3-4554-a143-e80504041d32)

**Comment by gegere** [comment]
The Tesla Battery is now hooked up via the high-voltage connectors to the inverters battery connection.

In the process of setting up the LilyGo Board.

I have uncommeted the following:
#define TESLA_MODEL_3Y_BATTERY
#define BYD_CAN          //Enable this line to emulate a "BYD Battery-Box 


Following [the Inverter: GoodWe guide](https://github.com/dalathegreat/Battery-Emulator/wiki/Inverter:-Goodwe).

**Comment by gegere** [comment]
Outstanding! Everything out of the gate is working, so excited! Follow [these instructions](https://github.com/dalathegreat/Battery-Emulator/wiki/Battery:-Tesla-Model-S-3-X-Y#example-integration) made it feel like a morning breeze! 

CAN_INVERTER_MISSING
WARNING
6/24/2025, 2:03:47 PM
Inverter not sending messages via CAN for the last 60 seconds. Check wiring!

**Comment by gegere** [comment]
The **CAN_INVERTER_MISSING** issue was resolved by installing a 120 Ohm resistor on the GoodWe Can connector. (where you plug the CAN_H & CAN_L to the inverter) 

An issue I didn't realize until now, is the GoodWe A-ES Series Inverter can't communicate with the battery. I can't run the inverter off the battery, which is odd because the Battery-Emulator is able to display all Canbus data from the TESLA_MODEL_3Y_BATTERY.

Tomorrow I am going to recompile with the following setting:
#define LFP_CHEMISTRY          //Tesla specific setting, enable this line to startup in LFP mode

**Comment by gegere** [comment]
I have enabled:
LFP_CHEMISTRY (the inverter is unable to communicate correctly with the battery)
AND 
DEBUG_VIA_WEB to dive deeper... [ip address/debug]

**Comment by gegere** [comment]
Happy to report a successful setup, this was the battery option I needed to select with the GoodWe SolarGo App.

![Image](https://github.com/user-attachments/assets/8b4bc539-dcc5-4e1d-aa6b-376864425207)

**Comment by gegere** [comment]
@dalathegreat can this be added to the [upported-inverters-list](https://github.com/dalathegreat/Battery-Emulator/wiki#supported-inverters-list)?

**Comment by dalathegreat** [comment]
Thanks for testing, added to the Wiki!

<img width="684" height="86" alt="Image" src="https://github.com/user-attachments/assets/233f6548-1aaa-447d-96b6-ceffbebc582b" />

**Comment by saab95aero1129-gif** [comment]
I also use Goodwe A-ES9600, but I can't get Goodwe to connect to Lilygo Can. How did you manage to do it successfully?

<img width="862" height="1898" alt="Image" src="https://github.com/user-attachments/assets/1fcc179c-1860-4625-ae56-0060a0b2d8a4" />
<img width="862" height="1898" alt="Image" src="https://github.com/user-attachments/assets/8bb1d1ce-8a81-4232-a58f-b00cb5fd5d25" />

<img width="1108" height="1477" alt="Image" src="https://github.com/user-attachments/assets/a6034f54-b9a7-4f17-bcc4-df6273a7b911" />

**Comment by gegere** [comment]
Over CanBus, where are your communication wires?

**Comment by saab95aero1129-gif** [comment]
Hi

I am currently using a *LilyGo T-2CAN* to bridge a *2021+ Tesla Model X
battery module* and a *GoodWe GW9600A-ES (A-ES) inverter*.

My goal is to emulate a BYD battery protocol so the inverter can recognize
it. However, the GoodWe App is currently unable to detect the BYD battery
settings.

I have a few questions regarding this setup:

   1.

   Do I need to add a *120-ohm terminal resistor* to the GoodWe CAN bus
   line?
   2.

   Are there any other hardware or software configurations I should
   double-check or pay attention to?

Any advice or troubleshooting tips would be greatly appreciated!

Thanks

Eason


Jason Gegere ***@***.***> 於 2026年6月29日週一 下午11:11寫道：

**Comment by gegere** [comment]
CAN termination: Make sure the CAN bus is terminated correctly. If the GoodWe and LilyGo are the only two devices on the bus, there should be 120 Ω termination at each end of the CAN bus. Many CAN transceivers already include one internally, so it's worth checking before adding another resistor.

I have had to repeat this process for the initial setup. To inverter needs to be completely powered down, including PV power.

**Comment by saab95aero1129-gif** [comment]
i will try it.
your system still work?
thanks
Eason

Jason Gegere ***@***.***> 於 2026年7月1日週三 上午5:12寫道：

**Comment by gegere** [comment]
Yes, it has been working for a solid year. The system has only gotten more refined.

**Comment by saab95aero1129-gif** [comment]
Hello, the battery inverter is now connected, but I haven't connected to PV yet. Have you ever tried grid to charge the battery?

<img width="1477" height="1108" alt="Image" src="https://github.com/user-attachments/assets/e01a64ea-94b7-4346-a925-0a479f46910f" />
<img width="862" height="1898" alt="Image" src="https://github.com/user-attachments/assets/1f502282-491f-4880-9b42-22c9ca5e2497" />
https://github.com/user-attachments/assets/f84a2164-20d7-4e45-bdc1-83650da28f20

<img width="1477" height="1108" alt="Image" src="https://github.com/user-attachments/assets/8457d760-c9d3-4741-a587-c7317dc4cdd8" />
